timely/src/progress/broadcast.rs

```diff
@@ -134,8 +134,8 @@ impl<T:Timestamp+Send> Progcaster<T> {
             // options for improving it if performance limits users who want other logging.
             self.progress_logging.as_ref().map(|l| {
 
-                let mut messages = Box::new(Vec::with_capacity(changes.len()));
-                let mut internal = Box::new(Vec::with_capacity(changes.len()));
+                let mut messages = Vec::with_capacity(changes.len());
+                let mut internal = Vec::with_capacity(changes.len());
```

Comment:
Note that we should be using `recv_changes.len()` here, because `changes` doesn't have any relation to what we're receiving. But, then we'd over-allocate :/

Reply:
Ah, could you explain that more? I agree that `recv_changes.len()` looks more appropriate, but what is the downside (seems like we fill these up with exactly that many elements, and then ship them in the `TimelyProgressEvent`)?